ipn_main_handler.php

```diff
@@ -467,9 +467,20 @@
       }
       // update order status history with new information
       ipn_debug_email('IPN NOTICE :: Set new status ' . $new_status . " for order ID = " .  $ordersID . ($_POST['pending_reason'] != '' ? '.   Reason_code = ' . $_POST['pending_reason'] : '') );
-      if ((int)$new_status == 0) $new_status = 1;
+      if ((int)$new_status == 0) {
+        $new_status = 1;
+        ipn_debug_email('IPN NOTICE :: Set new status ' . $new_status . " for order ID = " .  $ordersID . ($_POST['pending_reason'] != '' ? '.   Reason_code = ' . $_POST['pending_reason'] : '') );
+      }
       if (in_array($_POST['payment_status'], array('Refunded', 'Reversed', 'Denied', 'Failed'))
            || substr($txn_type,0,8) == 'cleared-' || $txn_type=='echeck-cleared' || $txn_type == 'express-checkout-cleared') {
+        $sql = "select orders_status from " . TABLE_ORDERS . "
+                 where orders_id = :ordersID:";
+        $sql = $db->bindVars($sql, ':ordersID:', $ordersID, 'integer');
+        $old_status = $db->Execute($sql);
+        if ($new_status < $oldstatus->fields['orders_status'] && (substr($txn_type, 0, 8) == 'cleared-' || $txn_type=='echeck-cleared' || $txn_type == 'express-checkout-cleared')) {
+          $new_status = $old_status->fields['orders_status'];
+        }
+        ipn_debug_email('IPN NOTICE :: Set new status ' . $new_status . " for order ID = " .  $ordersID . ($_POST['pending_reason'] != '' ? '.   Reason_code = ' . $_POST['pending_reason'] : '') );
```

Comment:
Between lines 469-483 we don't need to have these 3 calls to `ipn_debug_email`. 

Please remove the one on 469 and 472. The one on 483 will suffice.
And restore line 470 to its single-line setting of $new_status.